### ISSUE TITLE
Framework Dependent Publish doesn't work on  2.1.400

### ISSUE THREAD

**Issue by TatiTheFreaK** [dotnet/cli#9852]
## Steps to reproduce
Try to publish web api with `dotnet publish -f netcoreapp2.0 -c Release`, or within visual studio, with framework dependent selected

## Expected  behavior
Publish around 30 files

## Actual behavior
Publishing over 200 files (self-contained)

## Environment data
`dotnet --info` output:

SDK do .NET Core (refletindo qualquer global.json):
 Version:   2.1.400
 Commit:    8642e60a0f

Ambiente de tempo de execução:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.400\

Host (useful for support):
  Version: 2.1.2
  Commit:  811c3ce6c0

.NET Core SDKs installed:
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.302 [C:\Program Files\dotnet\sdk]
  2.1.400 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:

> https://aka.ms/dotnet-download

## Description

After upgrading to Visual Studio v15.8, which comes with .net core SDK v2.1.4, I've been trying to publish my WebAPI in a Framework Dependent way, but it only generates the Self Contained way, with over 200 files.

**Comment by livarcocc** [comment]
Can you share your csproj?

**Comment by TatiTheFreaK** [comment]
```<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Compile Remove="Controllers\ValuesController.cs" />
  </ItemGroup>

  <ItemGroup>
    <Folder Include="wwwroot\" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.5" />
    <PackageReference Include="Microsoft.AspNetCore.Cors" Version="2.0.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.0.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="2.0.1" />
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="2.0.2" />
    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="5.2.1" />
  </ItemGroup>

  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Tools" Version="2.0.2" />
  </ItemGroup>

  <ProjectExtensions><VisualStudio><UserProperties appsettings_1json__JSONSchema="" /></VisualStudio></ProjectExtensions>

</Project>```

**Comment by livarcocc** [comment]
@vijayrkn is this something that you can look at? Is this expected? Maybe because 2.1.400 does not have the store anymore?

**Comment by vijayrkn** [comment]
@livarcocc - This is happening on `dotnet publish -f netcoreapp2.0 -c Release`. Shouldn't this publish the non-standalone way by default since there is non RID publish. Why would store have impact in this case?

**Comment by livarcocc** [comment]
It is publishing non-stand alone still, but it is finding need to publish a lot more assets than before, notably, asp.net assets.

**Comment by vijayrkn** [comment]
In the actual behavior above, it is mentioned that it is publishing self contained 

```
Actual behavior
Publishing over 200 files (self-contained)
```

**Comment by TatiTheFreaK** [comment]
@vijayrkn actually, I'm not sure... I've searched about the cli publishing a high number of files and the results said that perhaps it was publishing in self contained mode. How can I verify? I know that there isn't an exe on the output folder.

**Comment by dasMulli** [comment]
Quite a few users have been confused over the meaning of "self-contained" meaning "a lot of DLL files" vs "contains runtime" both here on GH and on StackOverflow.
So @TatiTheFreaK does the output contain runtime assets - like `clrjit.dll`, `coreclr.dll` - or just a lot of asp.net core related dll files?

**Comment by TatiTheFreaK** [comment]
@dasMulli the output folder doesn't contain these files you mentioned

**Comment by vijayrkn** [comment]
@JunTaoLuo - Any changes on the ASP.NET side that could be causing this?

**Comment by dasMulli** [comment]
Looks like `PublishWithAspNetCoreTargetManifest` is a no-op in 2.1.400.
I can reproduce this with any asp.net core 2.0 app

[publish-2.1.302.binlog.zip](https://github.com/dotnet/cli/files/2298110/publish-2.1.302.binlog.zip)
[publish-2.1.400.binlog.zip](https://github.com/dotnet/cli/files/2298111/publish-2.1.400.binlog.zip)

**Comment by livarcocc** [comment]
also cc @natemcmaster in case he knows if this is expected. Also, the manifest target above, I believe that's for the store.

**Comment by JunTaoLuo** [comment]
Is the `SelfContained` property no longer set in 2.1.400? I was under the impression that both it and `UseAppHost` are set for back-compat purposes, i.e. exactly this scenario. In any case, I don't know how much we will prioritize this issue as 2.0 will soon go out of support on Oct 1, 2018 (https://www.microsoft.com/net/support/policy).

**Comment by natemcmaster** [comment]
This appears to be a regression in the 2.1.400 SDK. Even though the 2.0 runtime is going out of support, the SDK should still support building older TFMs. We cannot make changes to shipped NuGet packages, so we'll need a fix in Microsoft.NET.Sdk.

@TatiTheFreaK If upgrading to 2.1 is infeasible, try adding this to your .csproj file. In my minimal repro, this appears to workaround the issue.
```xml
<PropertyGroup>
  <PublishWithAspNetCoreTargetManifest>true</PublishWithAspNetCoreTargetManifest>
</PropertyGroup>
```

**Comment by dasMulli** [comment]
Yeah this is a slight but impactful change in the SDKs behavior / API.
Though the `SelfContained` property is properly set for back-compat when publishing self-contained, it is not explicitly set to `false` for portable applications, this failing the `== 'fasle'` check.

**Comment by TatiTheFreaK** [comment]
@natemcmaster I'll try to upgrade to 2.1 later, but adding these lines didn't work here.

**Comment by nguerrera** [comment]
So the fixfor compat  is  to make sure we set it to false and never leave it empty? Seems reasonable.

**Comment by nguerrera** [comment]
(MSBuild handling of booleans is baffling, btw. Empty string is not coerced to false in such a comparison, but 'no' and 'off' are.)

**Comment by dasMulli** [comment]
> MSBuild handling of booleans is baffling

I wouldn't be surprised if `'$(SelfContained)' &lt; 'true'` somehow magically worked.

**Comment by peterhuene** [comment]
@livarcocc servicing consideration for 2.1.4xx?  We need to restore setting `SelfContained` explicitly to false to ensure the websdk doesn't break here.

A much larger issue is ensuring regressions like this do not happen in the future.  MSBuild being so, ahem, flexible leads to seemingly innocuous changes causing breaks like this.  While I might argue that conditionals like the above should check for `!= 'true'` to support unset also meaning false, this is on me for only going so far as validating the change for the .NET Core SDK.

I will of course add test coverage for this break as part of the fix, but I think we should consider spending some cycles to shore up our test coverage that includes the websdk.

**Comment by livarcocc** [comment]
Yes, let's prepare a PR to fix this and get the ASK mode template filled out.

**Comment by dasMulli** [comment]
Are there any side effects of .400 (maybe also .401) having this bug?
e.g. if I publish once with .400 tooling to azure, then again with say a fixed .402 version, am I left with old runtime dll files that would theoretically downgrade the version of asp.net core if the OP updates from 2.0.5 to 2.0.8?

**Comment by peterhuene** [comment]
@TatiTheFreaK another workaround if you don't want to edit your project file is to publish with `dotnet publish -f netcoreapp2.0 -c Release -p:SelfContained=false`.  The bug is that `SelfContained` should be implicitly set to `false`, but it's actually an empty string with the 2.1.4xx SDK.   I have a fix ready that we hope to get into a 2.1.4xx servicing release.  Thanks very much for reporting this issue to us!

**Comment by vivmishra** [comment]
Add a known issue for 2.1.400

**Comment by peterhuene** [comment]
Unfortunately the fix didn't meet the servicing bar due to there being a reasonable (although, inconvenient) workaround.  This will be fully fixed in 3.0.
